src/Bicep.Core.UnitTests/Parsing/ParserTests.cs

```diff
@@ -246,9 +246,19 @@ public void PropertyAccessShouldParseSuccessfully(string text, string expected)
             RunExpressionTest(text, expected, typeof(PropertyAccessSyntax));
         }
 
+        [DataTestMethod]
+        [DataRow("a:b","(a:b)")]
+        [DataRow("null:fail", "(null:fail)")]
+        [DataRow("foo():bar","(foo():bar)")]
```

Comment:
Mind adding test(s) to assert the current behavior with the ternary operator just to avoid a potential future accidental regression?

Reply:
+1

---
In reply to: [585226540](https://github.com/Azure/bicep/pull/1516#discussion_r585226540) [](ancestors = 585226540)

Reply:
Added. This was a good callout, and I actually found a bug.